src/Parsers/ParserCreateQuery.h

```diff
@@ -199,6 +200,10 @@ bool IParserColumnDeclaration<NameParser>::parseImpl(Pos & pos, ASTPtr & node, E
         default_specifier = "EPHEMERAL";
         if (!expr_parser.parse(pos, default_expression, expected) && type)
             default_expression = std::make_shared<ASTLiteral>(Field());
+        /// The default expression for EPHEMERAL should be ASTLiteral,
+        /// otherwise it will be crash in formatImpl.
```

Comment:
This does not explain anything.

Reply:
It will treat `default_expression` for `EPHEMERAL` as `ASTLiteral` in `formatImpl`, but in these cases:
```sql
CREATE TABLE test(a UInt8, b EPHEMERAL String) Engine=Memory(); -- { clientError SYNTAX_ERROR }
CREATE TABLE test(a UInt8, b EPHEMERAL 'a' String) Engine=Memory(); -- { clientError SYNTAX_ERROR }
CREATE TABLE test(a UInt8, b String EPHEMERAL test) Engine=Memory(); -- { clientError SYNTAX_ERROR }
CREATE TABLE test(a UInt8, b String EPHEMERAL 1+2) Engine=Memory(); -- { clientError SYNTAX_ERROR }
```
it's not, so it's crash in here. 

https://github.com/ClickHouse/ClickHouse/blob/c8db250c88e8fd02383b1989c121f7c04b286f9e/src/Parsers/ASTColumnDeclaration.cpp#L76

Reply:
Updated. I think it's a proper fix.